### PR TITLE
Testing update

### PR DESCRIPTION
**Pull request by mwodahl**
# PR Details
## Description

This PR updates all references to the test package from org.junit.Test to org.junit.jupiter.api.Test. Some tests are also updated to remove deprecated annotations.

## Related Issue

Here's the link to the relevant issue for this PR: https://github.com/usdot-jpo-ode/jpo-ode/issues/450

## Motivation and Context

This change allows JUnit 5 to be used for testing and also allows mvn to discover and appropriately run all existing tests.

## How Has This Been Tested?

These changes have been tested by examining the output of mvn clean test. 

## Types of changes

- [ X ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[ODE Contributing Guide](https://github.com/usdot-jpo-ode/jpo-ode/blob/bugfix/Pull_request_template/docs/contributing_guide.md) 
- [ ] I have added tests to cover my changes.
- [ X ] All new and existing tests passed.

### CODE REVIEW

**Comment by dmccoystephenson** [comment]
> These changes have been tested by examining the output of mvn clean test.

It's possible this doesn't recompile the code itself, only the tests. Do you experience the same behavior when running `mvn clean package` followed by `mvn test`?

**Review by dmccoystephenson — APPROVED** [review]
The code looks good & the unit tests all pass! I'd wait for Brandon to review before merging, though.

File: jpo-ode-svcs/src/test/java/us/dot/its/jpo/ode/wrapper/MessageProducerTest.java
Comment (payneBrandon) [inline jpo-ode-svcs/src/test/java/us/dot/its/jpo/ode/wrapper/MessageProducerTest.java:55]: I see these were just ignored before and now disabled...do they not run? I realize we didn't really create or update them here but I'm curious why we have tests that aren't being used.

Comment (mwodahl) [reply jpo-ode-svcs/src/test/java/us/dot/its/jpo/ode/wrapper/MessageProducerTest.java:55]: These tests are skipped at the moment. I'm showing that 42 tests are being skipped when mvn test is run

File: jpo-ode-core/src/test/java/us/dot/its/jpo/ode/dds/CASClientTest.java
Comment (payneBrandon) [inline jpo-ode-core/src/test/java/us/dot/its/jpo/ode/dds/CASClientTest.java:430]:
```suggestion
               * surefire plug-in to report the followig very odd and strange errors on all
```

File: jpo-ode-core/src/test/java/us/dot/its/jpo/ode/dds/CASClientTest.java
Comment (payneBrandon) [inline jpo-ode-core/src/test/java/us/dot/its/jpo/ode/dds/CASClientTest.java:431]:
```suggestion
               * subsequent test cases. Hence, we have to use real values so we don't have to mock
```

File: jpo-ode-core/src/test/java/us/dot/its/jpo/ode/dds/CASClientTest.java
Comment (payneBrandon) [inline jpo-ode-core/src/test/java/us/dot/its/jpo/ode/dds/CASClientTest.java:467]: Same spelling updates here

**Review by payneBrandon — CHANGES_REQUESTED** [review]
A couple questions/comments

File: jpo-ode-core/src/test/java/us/dot/its/jpo/ode/dds/CASClientTest.java
Comment (mwodahl) [reply jpo-ode-core/src/test/java/us/dot/its/jpo/ode/dds/CASClientTest.java]: fixed spelling here

File: jpo-ode-core/src/test/java/us/dot/its/jpo/ode/dds/CASClientTest.java
Comment (mwodahl) [reply jpo-ode-core/src/test/java/us/dot/its/jpo/ode/dds/CASClientTest.java]: fixed spelling

File: jpo-ode-core/src/test/java/us/dot/its/jpo/ode/dds/CASClientTest.java
Comment (mwodahl) [reply jpo-ode-core/src/test/java/us/dot/its/jpo/ode/dds/CASClientTest.java]: Fixed spelling errors in these comments

**Review by payneBrandon — APPROVED** [review]

**Pull request merged**
